src/Microsoft.Health.Dicom.Core/Features/Validation/DicomElementMinimumValidator.cs

```diff
@@ -3,67 +3,53 @@
-            _byteBufferValidators.Add(DicomVR.SL, DicomElementMinimumValidation.ValidateSL);
-            _byteBufferValidators.Add(DicomVR.SS, DicomElementMinimumValidation.ValidateSS);
-            _byteBufferValidators.Add(DicomVR.UL, DicomElementMinimumValidation.ValidateUL);
-            _byteBufferValidators.Add(DicomVR.US, DicomElementMinimumValidation.ValidateUS);
-        }
-
-        // only works for single value dicom element
-        public void Validate(DicomElement element)
+            { DicomVR.AE, new DicomElementMaxLengthValidation(16) },
+            { DicomVR.AS, new DicomElementRequiredLengthValidation(4) },
+            { DicomVR.CS, new DicomElementMaxLengthValidation(16) },
+            { DicomVR.DA, new DicomDateValidation() },
+            { DicomVR.DS, new DicomElementMaxLengthValidation(16) },
+            { DicomVR.FL, new DicomElementRequiredLengthValidation(4) },
+            { DicomVR.FD, new DicomElementRequiredLengthValidation(8) },
+            { DicomVR.IS, new DicomElementMaxLengthValidation(12) },
+            { DicomVR.LO, new DicomLongStringValidation() },
+            { DicomVR.PN, new DicomPersonNameValidation() },
+            { DicomVR.SH, new DicomElementMaxLengthValidation(16) },
+            { DicomVR.SL, new DicomElementRequiredLengthValidation(4) },
+            { DicomVR.SS, new DicomElementRequiredLengthValidation(2) },
+            { DicomVR.UI, new DicomUidValidation() },
+            { DicomVR.UL, new DicomElementRequiredLengthValidation(4) },
+            { DicomVR.US, new DicomElementRequiredLengthValidation(2) },
+        };
+
+        public void Validate(DicomElement dicomElement)
         {
-            EnsureArg.IsNotNull(element, nameof(element));
-            DicomVR dicomVR = element.ValueRepresentation;
-
-            if (dicomVR == null)
+            EnsureArg.IsNotNull(dicomElement, nameof(dicomElement));
+            DicomVR vr = dicomElement.ValueRepresentation;
+            if (vr == null)
             {
                 Debug.Fail("Dicom VR type should not be null");
             }
-
-            if (_stringValidators.TryGetValue(dicomVR, out Action<string, string> stringValidator))
-            {
-                stringValidator(element.Get<string>(), GetName(element.Tag));
-            }
-            else if (_byteBufferValidators.TryGetValue(dicomVR, out Action<IByteBuffer, string> byteBufferValidator))
+            if (Validations.TryGetValue(vr, out IDicomElementValidation validationRule))
             {
-                byteBufferValidator(element.Buffer, GetName(element.Tag));
+                validationRule.Validate(dicomElement);
             }
             else
             {
-                Debug.Fail($"Missing validation action for for VR :{dicomVR.Code}, add a new validation and register in the constructor.");
+                Debug.Fail($"Validating VR {vr?.Code} is not supported.");
```

Comment:
I see we are using a lot of `Debug.Fail` in these APIs. Are these pre-conditions? Otherwise, should we be throwing errors instead?